src/binder/clrprivbindercoreclr.cpp

```diff
@@ -57,48 +58,58 @@ HRESULT CLRPrivBinderCoreCLR::BindAssemblyByName(IAssemblyName     *pIAssemblyNa
         ReleaseHolder<AssemblyName> pAssemblyName;
 
         SAFE_NEW(pAssemblyName, AssemblyName);
-        IF_FAIL_GO(pAssemblyName->Init(pIAssemblyName));
 
-        hr = BindAssemblyByNameWorker(pAssemblyName, &pCoreCLRFoundAssembly, false /* excludeAppPaths */);
-
-#if !defined(DACCESS_COMPILE) && !defined(CROSSGEN_COMPILE)
-        if ((hr == HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND)) ||
-            (hr == FUSION_E_APP_DOMAIN_LOCKED) || (hr == FUSION_E_REF_DEF_MISMATCH))
+        if (SUCCEEDED(pAssemblyName->Init(pIAssemblyName)))
         {
-            // If we are here, one of the following is possible:
-            //
-            // 1) The assembly has not been found in the current binder's application context (i.e. it has not already been loaded), OR
-            // 2) An assembly with the same simple name was already loaded in the context of the current binder but we ran into a Ref/Def
-            //    mismatch (either due to version difference or strong-name difference).
-            //
-            // Thus, if default binder has been overridden, then invoke it in an attempt to perform the binding for it make the call
-            // of what to do next. The host-overridden binder can either fail the bind or return reference to an existing assembly
-            // that has been loaded.
+            BinderTracing::ResolutionAttemptedOperation tracer{pAssemblyName};
 
-            // Attempt to resolve the assembly via managed TPA ALC instance if one exists
-            INT_PTR pManagedAssemblyLoadContext = GetManagedAssemblyLoadContext();
-            if (pManagedAssemblyLoadContext != NULL)
+            tracer.Trace(BinderTracing::ResolutionAttemptedOperation::Stage::FindInLoadContext);
```

Comment:
I think this will need to be deeper in the binder to differentiate between the `FindInLoadContext` and `PlatformAssemblies` stages (`AssemblyBinder::FindInExecutionContext` and `AssemblyBinder::BindByTpaList`)

Reply:
Will do this as a followup patch; this has taken way too long.